### PR TITLE
Control Display of Error, Info, Warning

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Adds `duration` property to `gr.Warning`, `gr.Info`, and `gr.Error`.

Setting `duration` to 0 not display the toast in the UI. This is useful for giving useful error messages for clients. For example, the data validation done in the component classes can now raise `gr.Error` instead of `TypeError` or `ValueError` and clients will see the exact error message.

The `duration` parameter is specified in seconds. If it's `None`, the toast will be displayed indefinitely.

Closes: https://github.com/gradio-app/gradio/issues/6480

modal_control

```python
import gradio as gr
from functools import partial

with gr.Blocks() as demo:
    with gr.Row():
        duration = gr.Number(label="Duration", info="Set to -1 for infinite", value=10, minimum=-1)
    with gr.Row():
        error = gr.Button("Error")
        info = gr.Button("Info")
        warning = gr.Button("Warning")

    def display_message(type, msg, duration):
        duration = None if duration < 0 else duration
        if type == "error":
            raise gr.Error(msg, duration=duration)
        elif type == "info":
            gr.Info(msg, duration=duration)
        elif type == "warning":
            gr.Warning(msg,  duration=duration)

    error.click(partial(display_message, "error", "ERROR 💥"),  [duration])
    info.click(partial(display_message, "info", "INFO ℹ️"), [duration])
    warning.click(partial(display_message, "warning", "WARNING ⚠️"), [duration])


demo.launch()
```

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/_website/src/lib/templates/gradio/05_modals/error.svx [hunk 1]
```diff
@@ -17,13 +17,19 @@
...
+raise gradio.Error("An error occured 💥!", duration=5)
 ```
 
 <!--- Description -->
 ### Description
 ## {@html style_formatted_text(obj.description)}
 
+## You can control for how long the error message is displayed with the `duration` parameter. If it's `None`, the message will be displayed forever until the user closes it. If it's a number, it will be shown for that many seconds. 
+
+## Below is a demo of how different values of duration control the error, info, and warning messages. You can see the code [here](https://huggingface.co/spaces/freddyaboulton/gradio-error-duration/blob/244331cf53f6b5fa2fd406ece3bf55c6ccb9f5f2/app.py#L17).
```
Comment (freddyaboulton) [inline js/_website/src/lib/templates/gradio/05_modals/error.svx:29]: @aliabd The link here is not being formatted as a link. How can I fix it?

image

Comment (aliabd) [reply js/_website/src/lib/templates/gradio/05_modals/error.svx:29]: pushed a commit that fixed it. also styling clash. 

what's happening with these issues is that i had overrided some default styling rules to design stuff (for example the side bar is a ul but we don't want bullet points there) and now we need them for the content.

Comment (freddyaboulton) [reply js/_website/src/lib/templates/gradio/05_modals/error.svx:29]: Thank you!

**Comment by abidlabs** [comment]
Btw I think this sufficiently closes https://github.com/gradio-app/gradio/issues/6480

> Provide a time parameter in seconds that would control the amount of time the message is displayed

This is complete

> Have a param provides the ability to position the message on the screen. Maybe allow a combination of top, middle, bottom, left, right, center. And/or potentially provide a control name reference, and have the message pop near that control. Would be great for field validation messages.

I don't think we want to do this, especially the second part as we are planning on instead having a dedicated way to do frontend validation

File: gradio/exceptions.py [hunk 2]
```diff
@@ -73,12 +75,18 @@ def divide(numerator, denominator):
     Demos: calculator, blocks_chained_events
     """
 
-    def __init__(self, message: str = "Error raised."):
+    def __init__(
+        self,
+        message: str = "Error raised.",
+        duration: int | None = 10,
```
Comment (abidlabs) [inline gradio/exceptions.py:81]: nit but 
```suggestion
        duration: int | float | None = 10,
```

**Comment by pngwn** [comment]
With number values I'm used to `0` meaning unlimited, although in this case 0 meaning hidden does make sense I think.

**Comment by aliabid94** [comment]
> With number values I'm used to 0 meaning unlimited, although in this case 0 meaning hidden does make sense I think.

I think it's just a bit confusing that 0 and None have such different meanings, when they're both "falsy". Wonder if another `visible=` kwarg would be more appropriate specifically for display.

File: client/js/src/helpers/api_info.ts [hunk 3]
```diff
@@ -318,6 +318,8 @@ export function handle_message(
 					status: {
 						queue,
 						message: data.output.error as string,
+						display: data.output.display as boolean,
```
Comment (aliabid94) [inline client/js/src/helpers/api_info.ts:321]: where is `display` passed from?

File: gradio/exceptions.py (hunk 2 above)
Comment (aliabid94) [reply gradio/exceptions.py:81]: just float is fine, int is a subset of float

File: gradio/queueing.py [hunk 4]
```diff
@@ -376,6 +381,7 @@ def log_message(
         event_id: str,
         log: str,
         level: Literal["info", "warning"],
+        duration: int | None = 10000,
```
Comment (aliabid94) [inline gradio/queueing.py:384]: why 10000 by default?

**Review by aliabid94 — APPROVED** [review]
just a few comments, otherwise LGTM!

**Comment by freddyaboulton** [comment]
Yes that's fair. Originally had a separate `display` kwarg but then thought that less parameters would be better. Will bring that back and clean up a bit to address the other comments. Thanks for the input all!

File: gradio/queueing.py (hunk 4 above)
Comment (freddyaboulton) [reply gradio/queueing.py]: Originally had it being set in ms and forgot to change the default. Fixed now.

File: client/js/src/helpers/api_info.ts (hunk 3 above)
Comment (freddyaboulton) [reply client/js/src/helpers/api_info.ts]: Originally had a separate display parameter. Added it back as `visible` so this makes more sense now.

**Pull request merged**